web_widget_many2many_binary_preview/static/src/web_widget_many2many_binary_preview.xml

```diff
@@ -0,0 +1,19 @@
+<templates>
+    <t
+        t-name="Many2ManyBinaryField.attachment_preview"
+        t-inherit="web.Many2ManyBinaryField.attachment_preview"
```

Comment:
Weird that the parent `attachment_preview` template does not in any way interface with the set of more advanced [attachment preview components](https://github.com/OCA/OCB/blob/16.0/addons/mail/static/src/components/attachment_image/attachment_image.xml#L3) that are in the `mail` module. Maybe that's a pending refactor that didn't make the 16.0 release.

Reply:
I'd almost think that it's better to make a new widget based on those components, since they have all kinds of [size logic](https://github.com/OCA/OCB/blob/16.0/addons/mail/static/src/models/attachment_image.js#L92) already. But that will probably open other cans of worms, so maybe just copy some of the logic to a new `getPreviewUrl` function and be done with it.

Reply:
> Maybe that's a pending refactor that didn't make the 16.0 release.

I don't [think](https://github.com/OCA/OCB/blob/18.0/addons/web/static/src/views/fields/many2many_binary/many2many_binary_field.xml#L28) so

> But that will probably open other cans of worms

None that I see, it will just be *much* more work to migrate this back and forth as the mail components have been refactored/rewritten a lot in the past releases, while the many2many_binary widget has barely changed.

I suggest for version 1 we just use the `/web/image` controller and expose its parameters (crop, width, height), that will already be very useful. Whenever inline preview should be implemented, we can look into using the mail components for that, but then we're already pretty much on [attachment_preview](https://odoo-community.org/shop/preview-attachments-531) territory and could just make that support images on many2many_binary fields.

Reply:
Agreed!